### PR TITLE
fix google login but need throw error

### PR DESCRIPTION
**Pull request by karael**

### CODE REVIEW

**Comment by jabbink** [comment]
Please remove the jitpack dependency. It should be unnecessary

**Comment by karael** [comment]
@jabbink you are right it's used in the lib, wait before accept my PR I have dependency problem

**Comment by pr0ves** [comment]
If you remove the jitpack dependecy, when building the bot, the build will fail because it won't find svarzee/gpsoauth on maven

**Comment by karael** [comment]
@pr0ves yep I have setup my intellij env this morning bug I m also working so I ll fix quickly as possible

**Comment by cq65617875** [comment]
e: /root/PokemonGoBot/src/main/kotlin/ink/abb/pogo/scraper/Main.kt: (33, 13): Unresolved reference: GoogleAutoCredentialProvider
:compileKotlin FAILED

**Comment by karael** [comment]
the problem was about the gpsoauth lib from svarzee. I solved the problem but I did some changes on the api lib

**Comment by wudustan** [comment]
So what's the latest regarding this? Still getting null from GoogleCredentialProvider.

**Comment by miracho** [comment]
8 Tem 15:48:00: Logging in to game server...
Exception in thread "main" java.lang.NoClassDefFoundError: svarzee/gps/gpsoauth/Gpsoauth$TokenRequestFailed
        at ink.abb.pogo.scraper.MainKt.getAuth(Main.kt:33)
        at ink.abb.pogo.scraper.MainKt.main(Main.kt:84)
Caused by: java.lang.ClassNotFoundException: svarzee.gps.gpsoauth.Gpsoauth$TokenRequestFailed
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 2 more
:run FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command 'C:\Program Files\Java\jdk1.8.0_101\bin\java.exe'' finished with non-zero exit value 1

**Comment by Skander1998** [comment]
@wudustan We unfortunately have to wait (or apply fix locally ourselves). :/

**Comment by megatixos** [comment]
How to fix locally ourselves?

**Comment by ApexZ** [comment]
I just missed out on a charizard rip me

**Comment by SiriusCrypto** [comment]
Any info on how to fix it ourselves?

File: src/main/kotlin/ink/abb/pogo/scraper/Main.kt [hunk 1]
```diff
@@ -30,16 +30,11 @@ fun getAuth(settings: Settings, http: OkHttpClient): CredentialProvider {
...
-                override fun onInitialOAuthComplete(googleAuthJson: GoogleAuthJson?) {
-                }
-
-                override fun onTokenIdReceived(googleAuthTokenJson: GoogleAuthTokenJson) {
-                    Log.normal("Setting Google refresh token in your config")
-                    settings.setToken(googleAuthTokenJson.refreshToken)
-                    settings.writeProperty("config.properties", "token")
-                }
-            }, time)
+            GoogleAutoCredentialProvider(http, username, password)
```
Comment (pr0ves) [inline src/main/kotlin/ink/abb/pogo/scraper/Main.kt]: Use `GoogleAutoCredentialProvider(http, username, password,time)` here, the function without time is deprecated.

**Comment by okynk** [comment]
@karael 
when I run with this gpsoauth fix, i got an exception when trying to login using google
```
Exception in thread "main" java.lang.StackOverflowError
	at java.lang.Character.codePointAt(Character.java:4866)
	at java.util.regex.Pattern$CharProperty.match(Pattern.java:3775)
	at java.util.regex.Pattern$Branch.match(Pattern.java:4604)
	at java.util.regex.Pattern$GroupHead.match(Pattern.java:4658)
	at java.util.regex.Pattern$LazyLoop.match(Pattern.java:4847)
	at java.util.regex.Pattern$GroupTail.match(Pattern.java:4717)
	at java.util.regex.Pattern$BranchConn.match(Pattern.java:4568)
	at java.util.regex.Pattern$CharProperty.match(Pattern.java:3777)
        .... so long trace same like above about regex.pattern .....
```
do you know why?

**Comment by megatixos** [comment]
@pr0ves can we fix this locally till global fix? and how!

**Comment by ApexZ** [comment]
so when will this be fixed???? xD

**Comment by ApexZ** [comment]
3 dragonites ive missed out on now can u fix plz

**Comment by BartWillems** [comment]
ApexZ oh well in that case, I'll notify the devs and let them know about all our dragonites we're missing out on!

**Comment by ApexZ** [comment]
Like they said they made a fix then why havent they released.. im confused with this communication

**Comment by karael** [comment]
there is a problem getting the gpsoauth lib, fixed soon

**Comment by ApexZ** [comment]
Ok do u have an estimate to when? @karael plz lmk!

**Comment by pr0ves** [comment]
@ApexZ, you have to calm down, we'll release the fix when is ready.

**Comment by ApexZ** [comment]
Ok its just I have work soon and wanted to catch some dragonites LOL

**Comment by ApexZ** [comment]
Rip https://pokevision.com/#/@37.80771146354739,-122.4219685792923 Missed out on a gyrados!

**Comment by BartWillems** [comment]
@ApexZ Quickly, go catch it before you have to go to work!

**Comment by ApexZ** [comment]
@BartWillems lol if pokebot was working lol.... xD

**Comment by wudustan** [comment]
ApexZ chill your shit. This is a voluntary project being ran by people who have other stuff to do. If you want to play the game go out and play.

**Comment by ApexZ** [comment]
any news?

**Comment by ApexZ** [comment]
@pr0ves @karael

**Comment by ApexZ** [comment]
my only question is will it possibly be fixed in the next 30 min or so

**Comment by karael** [comment]
@ApexZ won't fix it for a piece of shit like you

**Comment by ApexZ** [comment]
Ok im sorry ive been spamming its just I havent had time and I was really looking forward to today ok man no hard feelings @karael

**Comment by TunnelSnakesThatsus** [comment]
hello ive been seeming to get an error is there a fix yet?

**Comment by Rick058** [comment]
To the guy complaining: I made an account to tell you, and everyone else complaining, that you have no right to complain unless you are the ones that actual participate in developing the Pokémon Go Bot. Come on, they provide you and many others with a free working bot that is excellent. One day it does not work and you have to spam? No need mate, be grateful for what they do and wait patiently. Or go outside and catch the Pokémon yourself :^). They know there is a problem, now just patiently wait and come back in a few hours... Maybe they uploaded a fix, until then. Let them do the work, if they want to and have the time for it.

**Comment by TunnelSnakesThatsus** [comment]
Im sorry im new to github it just was out of curiousity

**Comment by TunnelSnakesThatsus** [comment]
@karael just wondering what is the problem currently that is causing this.

**Comment by kZahradnik** [comment]
Man,  all the spam doesn't help and makes the thread hard to read. Please stop it.

**Comment by pr0ves** [comment]
Removed the ot comments, here we'll discuss only about the problem and how to fix it.
@karael What changes did you do to the api?

**Comment by TunnelSnakesThatsus** [comment]
ok @pr0ves

**Comment by TunnelSnakesThatsus** [comment]
@pr0ves I heard about that manual fix did u ever try to see if it works?

**Comment by TunnelSnakesThatsus** [comment]
@pr0ves ??

**Comment by pr0ves** [comment]
Sorry @karael, I had to merge the fix asap.

**Comment by pr0ves** [comment]
Closing because the fix is merged with #575.

**Pull request closed without merging**